passes/opt/opt_expr.cc

```diff
@@ -745,6 +745,23 @@ void replace_const_cells(RTLIL::Design *design, RTLIL::Module *module, bool cons
 			}
 		}
 
+		if (cell->type == ID($shiftx) && cell->getPort(ID::Y).size() == 1) {
```

Comment:
I don't quite understand the reason for limiting to Y_WIDTH == 1. I think it should always be safe to trim `x` bits from the MSB end of A for `$shitfx`, and it should always be safe to trim `0` bits from the MSB end of A for `$shift`.

Nitpick: `GetSize(cell->getPort(ID::Y))` would be the canonical form for getting the size of Y.